devops: added buildbot to test Lambda on CI

mxschmitt opened the pull request:
Hello,

thought something like that would be cool to implement, docs are missing but waiting for the API to be fully clarified beforehand.

Also probably it makes more sense to move maybe Playwright and Puppeteer into the dependencies, because otherwise for users who are using Puppeteer it will say the peer dependency Playwright is missing and vice-versa.

Would be also cool if you could enable GitHub Actions on the repository, they are free and makes us possible to run CI on this repository.

(Tested on one of my repos which uses Vercel which uses AWS Lambda under the hood, works fine, see [here](https://github.com/playwright-community/playwright-community/pull/38/files#diff-541792fbd67c15c903b9b0280de5f01ed356d22500d46ef769dde9e4e8c3685b).)

aarvay commented:
I tried this one too since I had a similar use-case and seems to work for me too. It'd be great if this can be merged, @alixaxel !

mxschmitt commented:
Friendly ping @alixaxel and thank you for your impressive effort with the great Ansible build tooling kit there!

alixaxel commented:
Thanks for the PR @mxschmitt - GitHub Actions is something that I've been meaning to add support for for a while.

As for playwright, I'm sorry but I don't understand which use case this PR is trying to accomplish.
The reason why puppeteer is listed as a peer dependency has to do with 2 factors:

1) When this project started, only `puppeteer` was available, eventually we needed a way to also handle `puppeteer-core`.

https://github.com/alixaxel/chrome-aws-lambda/blob/d93ee402877b744a543adcbc7059e4991f801115/source/index.ts#L216-L224

2) There's some optional puppeteer-specific overloading going on:

https://github.com/alixaxel/chrome-aws-lambda/blob/d93ee402877b744a543adcbc7059e4991f801115/source/index.ts#L212-L214

Since this PR addresses none, I don't see what benefit it adds over simply doing the following:

```javascript
const chromium = require('chrome-aws-lambda');
const playwright = require('playwright-core');

(async () => {
  const browser = await playwright.chromium.launch({
    args: chromium.args,
    executablePath: await chromium.executablePath,
    headless: chromium.headless,
  });

  await browser.close();
})();
```

---

> Also probably it makes more sense to move maybe Playwright and Puppeteer into the dependencies, because otherwise for users who are using Puppeteer it will say the peer dependency Playwright is missing and vice-versa.

This would have the effects of installing both packages, which is undesirable due to size constraints.

Also with NPM 7, [peerDependencies are automatically installed](https://github.com/npm/rfcs/blob/latest/implemented/0025-install-peer-deps.md), so perhaps I will have to review this for `puppeteer` as well.

mxschmitt commented:
> Thanks for the PR @mxschmitt - GitHub Actions is something that I've been meaning to add support for for a while.
> 
> As for playwright, I'm sorry but I don't understand which use case this PR is trying to accomplish.
> The reason why puppeteer is listed as a peer dependency has to do with 2 factors:
> 
> 1) When this project started, only `puppeteer` was available, eventually we needed a way to also handle `puppeteer-core`.
> 
> https://github.com/alixaxel/chrome-aws-lambda/blob/d93ee402877b744a543adcbc7059e4991f801115/source/index.ts#L216-L224
> 
> 2) There's some optional puppeteer-specific overloading going on:
> 
> https://github.com/alixaxel/chrome-aws-lambda/blob/d93ee402877b744a543adcbc7059e4991f801115/source/index.ts#L212-L214
> 
> Since this PR addresses none, I don't see what benefit it adds over simply doing the following:
> 
> ```javascript
> const chromium = require('chrome-aws-lambda');
> const playwright = require('playwright-core');
> 
> (async () => {
>   const browser = await playwright.chromium.launch({
>     args: chromium.args,
>     executablePath: await chromium.executablePath,
>     headless: chromium.headless,
>   });
> 
>   await browser.close();
> })();
> ```
> 
> ---
> 
> > Also probably it makes more sense to move maybe Playwright and Puppeteer into the dependencies, because otherwise for users who are using Puppeteer it will say the peer dependency Playwright is missing and vice-versa.
> 
> This would have the effects of installing both packages, which is undesirable due to size constraints.
> 
> Also with NPM 7, [peerDependencies are automatically installed](https://github.com/npm/rfcs/blob/latest/implemented/0025-install-peer-deps.md), so perhaps I will have to review this for `puppeteer` as well.

Yeah I get you, so what do you think about:
- removing the Playwright part from the PR
- keeping the GitHub Action logic, since it's good to have integration tests
- adding an example how to use with Playwright to the readme (like you showed me)

Can also split that in multiple PRs if you prefer that.

alixaxel commented:
@mxschmitt I've already added a [playwright example](https://github.com/alixaxel/chrome-aws-lambda#usage-with-playwright) to the README. 🙂 

As for the GitHub Actions, they are welcome! 🚀

mxschmitt commented:
> @mxschmitt I've already added a [playwright example](https://github.com/alixaxel/chrome-aws-lambda#usage-with-playwright) to the README. 🙂
> 
> As for the GitHub Actions, they are welcome! 🚀

didn't see! then I will reference your project in [awesome-playwright](https://github.com/mxschmitt/awesome-playwright/) too!

Thank you so much and sorry for the inconvenience. Updated the PR.
